From 6fe951b03508ddaf54aec4197f2b7696c560dbe4 Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
Date: Wed, 17 Apr 2013 10:58:09 +0200
Subject: [PATCH 559/715] drm/rcar-du: Don't clean up mode setting if it
 hasn't been initialized

When no error cleanup is required return the error code directly. This
fixes a crash in the error path.

Signed-off-by: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
(cherry picked from commit 1034ba26a5ab26f519bcc8124b8f9c4e0f1c691f)

Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
(cherry picked from commit 781f03ba6ebda10d220936033c06b3a6fb7d8c98)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_drv.c |   12 ++++--------
 1 file changed, 4 insertions(+), 8 deletions(-)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_drv.c b/drivers/gpu/drm/rcar-du/rcar_du_drv.c
index 161913c..982f7d4 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_drv.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_drv.c
@@ -163,24 +163,21 @@ static int rcar_du_load(struct drm_device *dev, unsigned long flags)
 	mem = platform_get_resource(pdev, IORESOURCE_MEM, 0);
 	if (mem == NULL) {
 		dev_err(&pdev->dev, "failed to get memory resource\n");
-		ret = -EINVAL;
-		goto done;
+		return -EINVAL;
 	}
 
 	ioarea = devm_request_mem_region(&pdev->dev, mem->start,
 					 resource_size(mem), pdev->name);
 	if (ioarea == NULL) {
 		dev_err(&pdev->dev, "failed to request memory region\n");
-		ret = -EBUSY;
-		goto done;
+		return -EBUSY;
 	}
 
 	rcdu->mmio = devm_ioremap_nocache(&pdev->dev, ioarea->start,
 					  resource_size(ioarea));
 	if (rcdu->mmio == NULL) {
 		dev_err(&pdev->dev, "failed to remap memory resource\n");
-		ret = -ENOMEM;
-		goto done;
+		return -ENOMEM;
 	}
 
 	/* LVDS0 */
@@ -234,8 +231,7 @@ static int rcar_du_load(struct drm_device *dev, unsigned long flags)
 	rcdu->clock = devm_clk_get(&pdev->dev, NULL);
 	if (IS_ERR(rcdu->clock)) {
 		dev_err(&pdev->dev, "failed to get clock\n");
-		ret = -ENOENT;
-		goto done;
+		return -ENOENT;
 	}
 
 	rcdu->clock_du1 = clk_get(NULL, "rcar-du.1");
-- 
1.7.10.4

